test/ContainerTest.php

```diff
@@ -217,15 +219,24 @@ public function testMultipleCallsToExpirationSecondsAggregates()
         $this->container->bar = 'baz';
         $this->container->baz = 'bat';
         $this->container->bat = 'bas';
+        $currentTimestamp     = time();
         $this->container->setExpirationSeconds(3600);
         $this->container->setExpirationSeconds(1800, 'foo');
         $this->container->setExpirationSeconds(900, ['baz', 'bat']);
         $storage = $this->manager->getStorage();
         $metadata = $storage->getMetadata($this->container->getName());
-        $this->assertEquals($_SERVER['REQUEST_TIME'] + 1800, $metadata['EXPIRE_KEYS']['foo']);
-        $this->assertEquals($_SERVER['REQUEST_TIME'] + 900, $metadata['EXPIRE_KEYS']['baz']);
-        $this->assertEquals($_SERVER['REQUEST_TIME'] + 900, $metadata['EXPIRE_KEYS']['bat']);
-        $this->assertEquals($_SERVER['REQUEST_TIME'] + 3600, $metadata['EXPIRE']);
+        $this->assertEquals($currentTimestamp + 1800, $metadata['EXPIRE_KEYS']['foo']);
+        $this->assertEquals($currentTimestamp + 900, $metadata['EXPIRE_KEYS']['baz']);
+        $this->assertEquals($currentTimestamp + 900, $metadata['EXPIRE_KEYS']['bat']);
+        $this->assertEquals($currentTimestamp + 3600, $metadata['EXPIRE']);
+    }
+
+    public function testSettingExpirationSecondsUsesCurrentTime()
```

Comment:
Unfortunately, this test doesn't prove the fix.

To test, I reverted the change you provided (`s/time()/$_SERVER['REQUEST_TIME']/`), and ran just this test; it passed. I tried resetting the sleep and expiration seconds values as well, but they always passed. I also tried putting the assignment _prior_ to the `setExpirationSeconds()` method, and that did not work, either.

While I do not doubt that the fix is correct, we need an accurate way to test it that emulates what happens in a normal server-side web request so that we can ensure the fix is correct and will not break with future updates.